riscv/src/register/mstatus.rs

```diff
@@ -84,30 +93,75 @@ impl Mstatus {
         self.bits & (1 << 1) != 0
     }
 
+    /// Update Supervisor Interrupt Enable
+    ///
+    /// Note this updates the [`Mstatus`] value, but does not affect the mstatus
+    /// CSR itself. See [`set_sie`]/[`clear_sie`] to directly update the CSR.
+    #[inline]
+    pub fn update_sie(&self, sie: bool) -> Self {
```

Comment:
Somewhat of a bikeshed, but methods that set a bit in a bitfield without clearing the rest of the bitfield typically use the `modify_*` convention, e.g. in `pac` crates generated with `svd2rust`.

Reply:
In `cortex-m`, they use the `set_*` convention. I would align with them

Reply:
I started with set but the concern was it would be confusing vs the functions which act directly on the csr. Also there's a semantic difference as the csr functions use parameterleas set/clear for single bit fields whereas the update/modify/... on Mstatus takes a parameter.